### DISCUSSION TITLE
Release 2.0

### DISCUSSION THREAD

**Discussion by warrensbox** [warrensbox/terraform-switcher#440] [Releases]
I figured there are going to be breaking changes, should we release a major version.

**Comment by warrensbox** [comment]
There are overlapping work between @MatrixCrawler (#437 ) and @MatthewJohn (#434)?
How should we proceed with this?

**Reply by warrensbox** [reply]
Adding @yermulnik

**Reply by MatthewJohn** [reply]
Once/if you're happy with #437 and it's merged (somewhere, either release branch if you wanted to go to v2 or to master), I'd merge that branch and deal with the conflicts :)

**Comment by yermulnik** [comment]
> I figured there are going to be breaking changes, should we release a major version.

@warrensbox Which breaking changes are you envisioning?

These are what we merged since `v1.1.1` — none worth of a `major` version bump imho:
* [7f11304] go: bump github.com/hashicorp/go-version from 1.6.0 to 1.7.0
* [d579012] fix: Only extract terraform file from Terraform release archive (#433)
* [4c24d53] docs: Fix links to `install.sh` (#432)
* [e8d0167] go: bump golang.org/x/crypto from 0.22.0 to 0.23.0 (#428)
* [9c37902] go: bump golang.org/x/sys from 0.19.0 to 0.20.0 (#426)
* [ec2c9a1] docs: Streamline/fix docu site files and README (#419)
* [585a5eb] feat(goreleaser): Enable Release Changelog (#418)
* [a455c66] 154 feature request option to dry run (#416)

These are what we're working on at the moment I guess:
* #429 
  * Effectively it is to fix incorrect logic from `v1.1.0` — not sure it's worth of `v2.0.0`
* #435 
  * OpenTofu support — I'd say it's worth of `minor` version bump: `v1.2.0`
* #437 
  * This is a _feature_ — not sure it's worth of `v2.0.0`

**Comment by warrensbox** [comment]
2.0 will not release:

* #429 
  * Effectively it is to fix incorrect logic from `v1.1.0` — not sure it's worth of `v2.0.0`
* #435 
  * OpenTofu support — I'd say it's worth of `minor` version bump: `v1.2.0`
* #437 
  * This is a _feature_ — not sure it's worth of `v2.0.0`
  
 These will be released in 1.2

However, 2.0 will be release to depreciate old functions:
See comments by @MatthewJohn  in the source code `removed in v2.0.0`

Sometime in the future.

**Reply by yermulnik** [reply]
Gotcha. Thanks 👍🏻

**Comment by MatthewJohn** [comment]
Sorry, I’m away this weekend so having to reply via email.
If we remove TOML support for chdir, was this present prior to the last release?
Given some of the issues that it’s caused (and given it is technically a breaking change), worth releasing as a major change so people are more aware?
Im quite aware that the install script by default installs the latest version, so might cause issues.. but if they see it was a major release might be better PR

Matt

**Reply by yermulnik** [reply]
I'd go with minor as I 100% agree with https://github.com/warrensbox/terraform-switcher/discussions/420#discussioncomment-9624774
Quoting that here to improve your email experience @MatthewJohn:
> This bug has been out for awhile and nobody has complained a lot. Perhaps, people are not using the TOML file as much as we thought they would.
> 
> I say lets have TOML in the home directory for now (only). If someone opens an issue, we know that TOML is used beyond the home directory.
>
> @MatthewJohn please proceed with these changes. Otherwise your code looks pretty solid
